lib/DevServerManager.ts

```diff
@@ -1,48 +1,86 @@
-// @ts-nocheck
-const { logger } = require('@hubspot/local-dev-lib/logger');
-const { COMPONENT_TYPES } = require('./projects/structure');
-const { i18n } = require('./lang');
-const { promptUser } = require('./prompts/promptUtils');
-const { DevModeInterface } = require('@hubspot/ui-extensions-dev-server');
-const {
+import { logger } from '@hubspot/local-dev-lib/logger';
+import { Environment } from '@hubspot/local-dev-lib/types/Config';
+import {
+  COMPONENT_TYPES,
+  ComponentTypes,
+  Component,
+} from './projects/structure';
+import { i18n } from './lang';
+import { promptUser } from './prompts/promptUtils';
+import { DevModeInterface as UIEDevModeInterface } from '@hubspot/ui-extensions-dev-server';
+import {
   startPortManagerServer,
   stopPortManagerServer,
   requestPorts,
-} = require('@hubspot/local-dev-lib/portManager');
-const {
+} from '@hubspot/local-dev-lib/portManager';
+import {
   getHubSpotApiOrigin,
   getHubSpotWebsiteOrigin,
-} = require('@hubspot/local-dev-lib/urls');
-const { getAccountConfig } = require('@hubspot/local-dev-lib/config');
+} from '@hubspot/local-dev-lib/urls';
+import { getAccountConfig } from '@hubspot/local-dev-lib/config';
+import { ProjectConfig } from '../types/Projects';
 
 const i18nKey = 'lib.DevServerManager';
 
 const SERVER_KEYS = {
   privateApp: 'privateApp',
   publicApp: 'publicApp',
+} as const;
+
+type ServerKey = keyof typeof SERVER_KEYS;
+
+type DevServerInterface = {
+  // eslint-disable-next-line @typescript-eslint/no-unsafe-function-type
+  setup?: Function;
+  start?: (options: object) => Promise<void>;
+  fileChange?: (filePath: string, event: string) => Promise<void>;
+  cleanup?: () => Promise<void>;
+};
+
+type DevServer = {
+  componentType: ComponentTypes;
+  serverInterface: DevServerInterface;
+};
+
+type ComponentsByType = {
+  [key in ComponentTypes]: { [key: string]: Component };
```

Comment:
It looks like this is saying that _every_ component type will always be present in this object, which doesn't seem to be the case. I think you want `[key in ComponentTypes]?: { [key: string]: Component };`

Reply:
Done in [086af30](https://github.com/HubSpot/hubspot-cli/pull/1335/commits/086af30e65d2a8ef88ded01a818d1388441170dd)